repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeFilter.scala

```diff
@@ -19,7 +19,193 @@
+import org.slf4j.{LoggerFactory, MDC}
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.web.filter.DelegatingFilterProxy
+
+import scala.util.{Failure, Success}
+
+@Named("reposeFilter")
+class ReposeFilter @Inject()(@Value(ReposeSpringProperties.NODE.NODE_ID) nodeId: String,
+                             @Value(ReposeSpringProperties.CORE.REPOSE_VERSION) reposeVersion: String,
+                             optMetricsService: Optional[MetricsService],
+                             reposeFilterLoader: ReposeFilterLoader,
+                             containerConfigurationService: ContainerConfigurationService,
+                             tracer: Tracer,
+                             uriRedactionService: UriRedactionService,
+                             responseHeaderService: ResponseHeaderService
+                            )
+  extends DelegatingFilterProxy {
+
+  private val optMetricRegistry: Option[MetricRegistry] = Option(optMetricsService.orElse(null)).map(_.getRegistry)
+
+  override def initFilterBean(): Unit = {
+    ThisLogger.info("{} -- Initializing ReposeFilter", nodeId)
+    reposeFilterLoader.setServletContext(getServletContext)
+    ThisLogger.trace("{} -- initialized.", nodeId)
+  }
+
+  override def doFilter(request: ServletRequest, response: ServletResponse, chain: FilterChain): Unit = {
+    reposeFilterLoader.getFilterContextList match {
+      case Some(filterContextList) =>
+        ThisLogger.trace("ReposeFilter processing request...")
+        val startTime = System.currentTimeMillis
+        if (Option(request.asInstanceOf[HttpServletRequest].getHeader(TRACE_REQUEST)).isDefined) {
+          MDC.put(TRACE_REQUEST, "true")
+        }
+
+        val contentBodyReadLimit = containerConfigurationService.getContentBodyReadLimit
+        val requestBodyInputStream =
+          if (contentBodyReadLimit.isPresent) {
+            new LimitedReadInputStream(contentBodyReadLimit.get, request.getInputStream)
+          } else {
+            request.getInputStream
+          }
+
+        val wrappedResponse = new HttpServletResponseWrapper(response.asInstanceOf[HttpServletResponse], MUTABLE, MUTABLE)
+        val bufferedInputStream = new BufferedServletInputStream(requestBodyInputStream)
+
+        // Since getParameterMap may read the body, we must reset the InputStream so that we aren't stripping
+        // the body when form parameters are sent.
+        bufferedInputStream.mark(Integer.MAX_VALUE)
+        val paramaterMap = new HttpServletRequestWrapper(request.asInstanceOf[HttpServletRequest], bufferedInputStream).getParameterMap
+        bufferedInputStream.reset()
+
+        // Wrapping the request to reset the inputStream/Reader flag
+        val wrappedRequest = new HttpServletRequestWrapper(request.asInstanceOf[HttpServletRequest], bufferedInputStream)
+        wrappedRequest.setAttribute("http://openrepose.org/queryParams", paramaterMap)
```

Comment:
Who uses this? also it's not just query params. I'm pretty sure you'll get body params as well.

Reply:
HERP.